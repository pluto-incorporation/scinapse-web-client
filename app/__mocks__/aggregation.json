{
  "data": {
    "years": [
      { "year": 2008, "doc_count": 0 },
      { "year": 2009, "doc_count": 25 },
      { "year": 2010, "doc_count": 14 },
      { "year": 2011, "doc_count": 13 },
      { "year": 2012, "doc_count": 13 },
      { "year": 2013, "doc_count": 12 },
      { "year": 2014, "doc_count": 13 },
      { "year": 2015, "doc_count": 0 },
      { "year": 2016, "doc_count": 0 },
      { "year": 2017, "doc_count": 0 },
      { "year": 2018, "doc_count": 0 }
    ],
    "journals": [
      { "id": 25293849, "title": "The FASEB Journal", "doc_count": 2 },
      { "id": 107775873, "title": "Hydrobiologia", "doc_count": 2 },
      { "id": 3880285, "title": "Science", "doc_count": 1 },
      { "id": 11380747, "title": "Canadian Veterinary Journal-revue Veterinaire Canadienne", "doc_count": 1 },
      { "id": 11973042, "title": "Plant Archives", "doc_count": 1 },
      { "id": 12065667, "title": "Mikrochimica Acta", "doc_count": 1 },
      { "id": 13275601, "title": "American Scholar", "doc_count": 1 },
      { "id": 20845397, "title": "Journal of The Canadian Dental Association", "doc_count": 1 },
      { "id": 24807848, "title": "Physical Review Letters", "doc_count": 1 },
      { "id": 26277462, "title": "International Journal of Agriculture and Biology", "doc_count": 1 }
    ],
    "impact_factors": [
      { "from": 0, "to": 1, "doc_count": 9 },
      { "from": 1, "to": 2, "doc_count": 9 },
      { "from": 2, "to": 3, "doc_count": 7 },
      { "from": 3, "to": 4, "doc_count": 2 },
      { "from": 4, "to": 5, "doc_count": 2 },
      { "from": 5, "to": 6, "doc_count": 2 },
      { "from": 6, "to": 7, "doc_count": 0 },
      { "from": 7, "to": 8, "doc_count": 0 },
      { "from": 8, "to": 9, "doc_count": 1 },
      { "from": 9, "to": 10, "doc_count": 0 },
      { "from": 10, "to": null, "doc_count": 32 }
    ],
    "fos_list": [
      { "id": 127413603, "name": "Engineering", "doc_count": 33 },
      { "id": 71924100, "name": "Medicine", "doc_count": 29 },
      { "id": 185592680, "name": "Chemistry", "doc_count": 24 },
      { "id": 41008148, "name": "Computer Science", "doc_count": 22 },
      { "id": 121332964, "name": "Physics", "doc_count": 21 },
      { "id": 86803240, "name": "Biology", "doc_count": 20 },
      { "id": 192562407, "name": "Materials Science", "doc_count": 14 },
      { "id": 17744445, "name": "Political Science", "doc_count": 13 },
      { "id": 33923547, "name": "Mathematics", "doc_count": 12 },
      { "id": 46141821, "name": "Nuclear magnetic resonance", "doc_count": 9 }
    ]
  }
}
