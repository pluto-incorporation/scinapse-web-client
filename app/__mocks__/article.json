{
  "id": 20,
  "type": "POST_PAPER",
  "title": "this is title",
  "articleAbstract": "this is abstract.",
  "summary": "this is summary.",
  "link": "this is summary.",
  "source": "ARXIV",
  "articlePublishedAt": "2017-10-10T10:10:10",
  "articleUpdatedAt": "2017-10-10T10:10:10",
  "point": {
    "total": 7,
    "originality": 7,
    "contribution": 7,
    "analysis": 7,
    "expressiveness": 7
  },
  "createdBy": {
    "id": 10,
    "email": "test",
    "name": "tedyeee",
    "reputation": 16
  },
  "createdAt": "2017-10-09T10:47:32.105",
  "authors": [
    {
      "id": 30,
      "type": "LEAD_AUTHOR",
      "name": "pepsi",
      "organization": "Coca Cola"
    },
    {
      "id": 31,
      "member": {
        "id": 10,
        "email": "test",
        "name": "tedyeee",
        "reputation": 16
      },
      "type": "CORRESPONDING_AUTHOR",
      "name": "coke",
      "organization": "kingsi"
    }
  ],
  "evaluations": [
    {
      "id": 50,
      "point": {
        "total": 7,
        "originality": 7,
        "contribution": 7,
        "analysis": 7,
        "expressiveness": 7,
        "originalityComment": "originality comment",
        "contributionComment": "contribution comment",
        "analysisComment": "analysis comment",
        "expressivenessComment": "expressiveness comment"
      },
      "vote": 1,
      "comments": [
        {
          "id": 60,
          "createdBy": {
            "id": 10,
            "email": "test",
            "name": "tedyeee",
            "reputation": 16
          },
          "comment": "this is comment.",
          "createdAt": "2017-10-09T10:49:26.719"
        }
      ],
      "createdBy": {
        "id": 10,
        "email": "test",
        "name": "tedyeee",
        "reputation": 16
      },
      "createdAt": "2017-10-09T10:48:22.298"
    }
  ]
}
