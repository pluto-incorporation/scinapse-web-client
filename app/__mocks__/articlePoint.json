{ "total": 6.75, "originality": 4.0, "significance": 6.0, "validity": 8.0, "organization": 9.0 }
