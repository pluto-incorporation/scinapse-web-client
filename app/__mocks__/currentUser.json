{
  "affiliation": "eeee",
  "articleCount": 0,
  "email": "honeyjasdds@postech.ac.kr",
  "emailVerified": false,
  "id": "39",
  "isLoggedIn": true,
  "major": null,
  "name": "academey",
  "oauth": null,
  "oauthLoggedIn": true,
  "profileImage": null,
  "reputation": 0,
  "reviewCount": 0
}
