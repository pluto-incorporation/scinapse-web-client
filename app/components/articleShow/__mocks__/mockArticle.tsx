export function getMockArticle() {
  return {
    articleId: 1,
    tags: ["Open Access Paper", "CAU Paper"],
  };
}
