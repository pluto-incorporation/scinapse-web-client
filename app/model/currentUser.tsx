import { Member } from "./member";

export interface CurrentUser
  extends Member,
    Readonly<{
      isLoggedIn: boolean;
      oauthLoggedIn: boolean;
    }> {}

export const CURRENT_USER_INITIAL_STATE: CurrentUser = {
  isLoggedIn: false,
  oauthLoggedIn: false,
  email: "",
  firstName: "",
  lastName: "",
  id: 0,
  profileImage: "",
  profileId: null,
  affiliation: "",
  major: "",
  commentCount: 0,
  emailVerified: false,
  oauth: null,
};
