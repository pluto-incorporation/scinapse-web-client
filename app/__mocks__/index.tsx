import { IArticle, recordifyArticle } from "../model/article";
import { IWallet, WalletFactory } from "../model/wallet";
import { IAuthor, recordifyAuthor } from "../model/author";
import { IMember, recordifyMember } from "../model/member";
import { IArticlePoint, ArticlePointFactory } from "../model/articlePoint";
import { IComment, recordifyComment } from "../model/comment";
import { IReviewPoint, ReviewPointFactory } from "../model/reviewPoint";
import { IReview, recordifyReview } from "../model/review";

export const RAW = {
  ARTICLE: require("./article.json") as IArticle,
  ARTICLE_POINT: require("./articlePoint.json") as IArticlePoint,
  AUTHOR: require("./author.json") as IAuthor,
  COMMENT: require("./comment.json") as IComment,
  REVIEW: require("./review.json") as IReview,
  REVIEW_POINT: require("./reviewPoint.json") as IReviewPoint,
  MEMBER: require("./member.json") as IMember,
  WALLET: require("./wallet.json") as IWallet,
};

export const RECORD = {
  ARTICLE: recordifyArticle(RAW.ARTICLE),
  ARTICLE_POINT: ArticlePointFactory(RAW.ARTICLE_POINT),
  AUTHOR: recordifyAuthor(RAW.AUTHOR),
  COMMENT: recordifyComment(RAW.COMMENT),
  REVIEW: recordifyReview(RAW.REVIEW),
  REVIEW_POINT: ReviewPointFactory(RAW.REVIEW_POINT),
  MEMBER: recordifyMember(RAW.MEMBER),
  WALLET: WalletFactory(RAW.WALLET),
};
