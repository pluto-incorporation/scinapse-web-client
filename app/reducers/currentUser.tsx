import { IReduxAction } from "../typings/actionType";
import { ACTION_TYPES } from "../actions/actionTypes";
import { CURRENT_USER_INITIAL_STATE, ICurrentUserRecord, CurrentUserFactory } from "../model/currentUser";

export function reducer(state = CURRENT_USER_INITIAL_STATE, action: IReduxAction<any>): ICurrentUserRecord {
  switch (action.type) {
    case ACTION_TYPES.SIGN_IN_SUCCEEDED_TO_SIGN_IN: {
      return CurrentUserFactory(action.payload.user).set("isLoggedIn", true);
    }

    case ACTION_TYPES.AUTH_SUCCEEDED_TO_SIGN_OUT: {
      return CURRENT_USER_INITIAL_STATE;
    }

    case ACTION_TYPES.AUTH_SUCCEEDED_TO_CHECK_LOGGED_IN: {
      return CurrentUserFactory(action.payload.user).set("isLoggedIn", true);
    }

    default:
      return state;
  }
}
