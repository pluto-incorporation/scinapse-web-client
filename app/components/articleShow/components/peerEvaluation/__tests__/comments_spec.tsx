jest.unmock("../comments");

import * as React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import EvaluationComments, { IEvaluationCommentsProps } from "../comments";
import { RECORD } from "../../../../../__mocks__";
import { CURRENT_USER_INITIAL_STATE } from "../../../../../model/currentUser";

describe("<EvaluationComments /> component", () => {
  let commentsWrapper: ShallowWrapper<IEvaluationCommentsProps>;

  beforeEach(() => {
    commentsWrapper = shallow(
      <EvaluationComments
        currentUser={CURRENT_USER_INITIAL_STATE}
        handlePeerEvaluationCommentSubmit={() => {}}
        evaluation={RECORD.EVALUATION}
      />,
    );
  });

  it("should match snapshot", () => {
    expect(commentsWrapper.html()).toMatchSnapshot();
  });
});
