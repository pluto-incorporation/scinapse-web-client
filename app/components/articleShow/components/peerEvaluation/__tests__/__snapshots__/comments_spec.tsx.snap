// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EvaluationComments /> component should match snapshot 1`] = `"<div class=\\"commentsWrapper\\"><div class=\\"commentWrapper\\"><div class=\\"authorInformation\\"><img class=\\"icon avatarIcon\\" src=\\"https://dd2gn9pwu61vr.cloudfront.net/avatar\\" alt=\\"AVATAR\\"/><span class=\\"authorName\\">Antonia Ruprecht</span></div><div class=\\"commentArea\\">Trying to figure out the potential impact of Blockchain implementations in the setting of clinical trials will shed new light on how modern clinical trial methods could evolve and benefit from Blockchain technologies in order to tackle the aforementioned challenges.</div><div class=\\"commentedAt\\">6 days ago</div></div><div class=\\"commentWrapper\\"><div class=\\"authorInformation\\"><img class=\\"icon avatarIcon\\" src=\\"https://dd2gn9pwu61vr.cloudfront.net/avatar\\" alt=\\"AVATAR\\"/><span class=\\"authorName\\">Antonia Ruprecht</span></div><div class=\\"commentArea\\">Trying to figure out the potential impact of Blockchain implementations in the setting of clinical trials will shed new light on how modern clinical trial methods could evolve and benefit from Blockchain technologies in order to tackle the aforementioned challenges.</div><div class=\\"commentedAt\\">6 days ago</div></div><div class=\\"commentWrapper\\"><div class=\\"authorInformation\\"><img class=\\"icon avatarIcon\\" src=\\"https://dd2gn9pwu61vr.cloudfront.net/avatar\\" alt=\\"AVATAR\\"/><span class=\\"authorName\\">Antonia Ruprecht</span></div><div class=\\"commentArea\\">Trying to figure out the potential impact of Blockchain implementations in the setting of clinical trials will shed new light on how modern clinical trial methods could evolve and benefit from Blockchain technologies in order to tackle the aforementioned challenges.</div><div class=\\"commentedAt\\">6 days ago</div></div></div>"`;
